compiler/src/iree/compiler/Codegen/SPIRV/SPIRVTile.cpp

```diff
@@ -133,9 +133,10 @@ static LogicalResult tileAndDistributeToThreads(linalg::LinalgOp consumerOp,
   // after bufferization. So add attributes to the tiled loop nest to
   // indicate that they should be distributed to invocations.
   ArrayRef<scf::ForOp> loops = tileAndFuseResult.value().loops;
-  assert(loops.size() <= kNumGPUDims);
   const char *attrName = getSPIRVDistributeAttrName();
-  for (int i = loops.size() - 1, dim = 0; i >= 0; --i) {
+  // We can have more than 3 dimensions being tiled (e.g., for convolutions with
+  // non-1 batch). But only the innermost 3 dimensions are distributed.
+  for (int i = loops.size() - 1, dim = 0; i >= 0 && dim < 3; --i) {
```

Comment:
Could we rewrite this without so much manual indexing? E.g.: `for (auto [loop, dim] : zip(reverse(loops), seq(0, 3))`

Reply:
Nice, thanks for the suggestion!